src/app/editors/contentlist/contentlist.component.ts

```diff
@@ -1,137 +1,216 @@
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    },
-    {
-      name: 'collection',
-      contentType: 'course',
-      status: 'Draft'
-    },
-    {
-      name: 'textBook name',
-      contentType: 'TextBook',
-      status: 'Review'
-    },
-    {
-      name: 'collection name',
-      contentType: 'collection',
-      status: 'Draft'
-    }
-  ];
-  public currentPage: any;
-  constructor(private router: Router, private location: Location, public activatedRoute: ActivatedRoute) { }
+  public contentType: string;
+  public contentArray = [];
+  public editorConfig = {};
+  public content: any;
+  public hierarchyConfig = {};
+  public channelData: any;
+  public apiErrorResponse: any;
+  constructor(private router: Router, private location: Location, public activatedRoute: ActivatedRoute,
+              public helperService: HelperService) { }
 
   ngOnInit(): void {
     this.activatedRoute.params
       .subscribe(
         (params) => {
-          console.log(params);
           this.contentType = params['page'];
         }
       );
+    this.getAllCollectionList();
   }
-  navigateToContentCreate() {
-    this.router.navigate(['editors/' + this.contentType]);
-  }
+
   goBack() {
     this.location.back();
   }
+  getContentDetails(id) {
+if (id) {
+  this.openContent(id);
+}
+  }
+  openContent(id) {
+    this.helperService.getContent(id)
```

Comment:
No logic to check enter data is valid string and should have trim other checks before proceeding to api call